File: SeleniumTest.java
// File: SeleniumTest.java
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.openqa.selenium.firefox.FirefoxOptions;

public class SeleniumTest {
    public static void main(String[] args) {
        // Test with Chrome
        testChrome();
        // Test with Firefox
        testFirefox();
    }

    public static void testChrome() {
        System.out.println("Testing with Chrome...");

        // Set ChromeDriver path
        System.setProperty("webdriver.chrome.driver", "/usr/local/bin/chromedriver");

        // Initialize ChromeDriver
        ChromeOptions chromeOptions = new ChromeOptions();
        // Uncomment the line below to run Chrome in headless mode
        // chromeOptions.addArguments("--headless");
        WebDriver driver = new ChromeDriver(chromeOptions);

        try {
            // Navigate to Google
            driver.get("https://www.google.com");
            System.out.println("Chrome - Page Title: " + driver.getTitle());
        } finally {
            // Close the browser
            driver.quit();
        }
    }

    public static void testFirefox() {
        System.out.println("Testing with Firefox...");

        // Set GeckoDriver path
        System.setProperty("webdriver.gecko.driver", "/usr/local/bin/geckodriver");

        // Initialize GeckoDriver
        FirefoxOptions firefoxOptions = new FirefoxOptions();
        // Uncomment the line below to run Firefox in headless mode
        // firefoxOptions.addArguments("--headless");
        WebDriver driver = new FirefoxDriver(firefoxOptions);

        try {
            // Navigate to Google
            driver.get("https://www.google.com");
            System.out.println("Firefox - Page Title: " + driver.getTitle());
        } finally {
            // Close the browser
            driver.quit();
        }
    }
}

File: selenium_test.js
// File: selenium_test.js
const { Builder } = require('selenium-webdriver');
const chrome = require('selenium-webdriver/chrome');
const firefox = require('selenium-webdriver/firefox');

const CHROME_DRIVER_PATH = '/usr/local/bin/chromedriver';
const FIREFOX_DRIVER_PATH = '/usr/local/bin/geckodriver';

async function testChrome() {
    console.log('Testing with Chrome...');
    let driver;
    try {
        // Set up Chrome options
        const chromeOptions = new chrome.Options();
        // Uncomment the line below to run Chrome in headless mode
        // chromeOptions.addArguments('--headless');

        // Initialize ChromeDriver
        driver = await new Builder()
            .forBrowser('chrome')
            .setChromeOptions(chromeOptions)
            .setChromeService(new chrome.ServiceBuilder(CHROME_DRIVER_PATH))
            .build();

        // Navigate to Google
        await driver.get('https://www.google.com');
        const title = await driver.getTitle();
        console.log('Chrome - Page Title:', title);
    } finally {
        // Close the browser
        if (driver) await driver.quit();
    }
}

async function testFirefox() {
    console.log('Testing with Firefox...');
    let driver;
    try {
        // Set up Firefox options
        const firefoxOptions = new firefox.Options();
        // Uncomment the line below to run Firefox in headless mode
        // firefoxOptions.addArguments('--headless');

        // Initialize GeckoDriver
        driver = await new Builder()
            .forBrowser('firefox')
            .setFirefoxOptions(firefoxOptions)
            .setFirefoxService(new firefox.ServiceBuilder(FIREFOX_DRIVER_PATH))
            .build();

        // Navigate to Google
        await driver.get('https://www.google.com');
        const title = await driver.getTitle();
        console.log('Firefox - Page Title:', title);
    } finally {
        // Close the browser
        if (driver) await driver.quit();
    }
}

(async () => {
    // Run tests for both browsers
    await testChrome();
    await testFirefox();
})();